### PR TITLE
chore: fixes to tests for JAX 0.5 upgrade

### PR DESCRIPTION
**Pull request by sash-a**
Jax 0.5 changed the output of random calls, so the same seed would output different numbers. Thus I needed to change some of the tests as they relied on the state of the random generation of the environment. 

Some of the envs had bad tests that tested the exact values of the state/obs I have changed these to check for shape/dtype/min/max instead

Closes: #277

### CODE REVIEW

File: jumanji/environments/logic/minesweeper/env_test.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def play_and_get_episode_stats(
     time_limit: int,
     force_start_state: Optional[State] = None,
 ) -> Tuple[List[float], List[StepType], int]:
-    state, timestep = jax.jit(env.reset)(jax.random.PRNGKey(0))
+    state, timestep = jax.jit(env.reset)(jax.random.PRNGKey(1))
```
Comment (WiemKhlifi) [inline jumanji/environments/logic/minesweeper/env_test.py:41]: Is there a reason for changing the seed? Or is it related to some test values?

File: jumanji/environments/routing/multi_cvrp/env_test.py [hunk 2]
```diff
@@ -230,8 +226,11 @@ def test_multicvrp__state_to_observation_timestep(self, multicvrp_env: MultiCVRP
         )
 
         # Check that the node demands and action masks are correct.
-        assert np.array_equal(obs.nodes.demands, test_node_demand)
-        assert np.array_equal(obs.action_mask, test_action_mask)
+        assert obs.nodes.demands.shape == (multicvrp_env._num_customers + 1,)
+        assert obs.action_mask.shape == (
+            multicvrp_env._num_vehicles,
```
Comment (WiemKhlifi) [inline jumanji/environments/routing/multi_cvrp/env_test.py:231]: 🧠 💯

File: jumanji/environments/routing/maze/env_test.py [hunk 3]
```diff
@@ -148,23 +148,29 @@ def test_maze__step(self, maze: Maze) -> None:
         assert state.agent_position == Position(row=2, col=2)
 
     def test_maze__action_mask(self, maze: Maze) -> None:
-        key = jax.random.PRNGKey(0)
-        state, _ = maze.reset(key)
-
+        walls = jnp.array(
```
Comment (WiemKhlifi) [inline jumanji/environments/routing/maze/env_test.py:151]: Is the reason for this to increase speed? I know the goal is to check action mask so initing the env is useless 🤔

**Review by WiemKhlifi — COMMENTED** [review]
Thanks @sash-a for fixing it! LGTM just have some comments 🙌

File: jumanji/environments/routing/maze/env_test.py (hunk 3 above)
Comment (sash-a) [reply jumanji/environments/routing/maze/env_test.py:151]: Similar to the other one, because 0.5 changes the output of seeds the reset produces a different state. These walls are just copying the reset walls from 0.4 JAX

**Review by WiemKhlifi — APPROVED** [review]
LGTM thanks @sash-a for these fixes 🥳
